modules/auxiliary/scanner/couchdb/couchdb_enum.rb

```diff
@@ -16,32 +16,93 @@ def initialize(info = {})
       },
       'References'     =>
         [
-          ['URL', 'https://wiki.apache.org/couchdb/HTTP_database_API']
+          ['URL', 'https://wiki.apache.org/couchdb/HTTP_database_API'],
+          ['CVE','2017-12635'],
+          ['URL','https://cve.mitre.org/cgi-bin/cvename.cgi?name=2017-12635'],
+          ['URL','https://justi.cz/security/2017/11/14/couchdb-rce-npm.html']
         ],
-      'Author'         => [ 'Roberto Soares Espreto <robertoespreto[at]gmail.com>' ],
+      'Author'         => [ 'Roberto Soares Espreto <robertoespreto[at]gmail.com>',
+                            'Hendrik Van Belleghem - @hendrikvb', # Database dump enhancements
+                            'Green-m <greenm.xxoo[at]gmail.com>' # Portions from apache_couchdb_cmd_exec.rb used
+                          ],
       'License'        => MSF_LICENSE
     ))
 
     register_options(
       [
-        Opt::RPORT(5984),
         OptString.new('TARGETURI', [true, 'Path to list all the databases', '/_all_dbs']),
         OptBool.new('SERVERINFO', [true, 'Print server info']),
-        OptString.new('HttpUsername', [false, 'The username to login as']),
-        OptString.new('HttpPassword', [false, 'The password to login with'])
+        OptBool.new('CREATEUSER', [true, 'Create Administrative user - ']),
+        OptString.new('HttpUsername', [true, 'CouchDB Username', Rex::Text.rand_text_alpha(12,"")]),
+        OptString.new('HttpPassword', [true, 'CouchDB Password', 'password']),
+        OptString.new('RPORT', [true, 'CouchDB Port', '5984']),
+        OptString.new('RHOST', [true, 'CouchDB Host', '']),
+        OptString.new('ROLES', [true, 'CouchDB Roles', '_admin'])
+
       ])
   end
 
   def valid_response(res)
     return res.code == 200 && res.headers['Server'].include?('CouchDB')
   end
 
+  def get_version
+    @version = nil
+
+    begin
+      res = send_request_cgi(
+        'uri'           => normalize_uri("/"),
+        'method'        => 'GET',
+        'authorization' => @auth
+      )
+    rescue Rex::ConnectionError
+      vprint_bad("#{peer} - Connection failed")
+      return false
+    end
+
+    unless res
+      vprint_bad("#{peer} - No response, check if it is CouchDB. ")
+      return false
+    end
+
+    if res && res.code == 401
+      print_bad("#{peer} - Authentication required.")
+      return false
+    end
+
+    if res && res.code == 200
+      res_json = res.get_json_document
+
+      if res_json.empty?
+        vprint_bad("#{peer} - Cannot parse the response, seems like it's not CouchDB.")
+        return false
+      end
+
+      @version = res_json['version'] if res_json['version']
+      return true
+    end
+
+    vprint_warning("#{peer} - Version not found")
+    return true
+  end
+
+  def check
+    get_version
+    version = Gem::Version.new(@version)
+    return CheckCode::Unknown if version.version.empty?
```

Comment:
This method does not work because `CheckCode` is undefined in an aux context. You need `Exploit::CheckCode`. Resolved myself.